test/topology_experimental_raft/test_tablets.py

```diff
@@ -33,18 +34,37 @@ async def inject_error_on(manager, error_name, servers):
     await asyncio.gather(*errs)
 
 
-async def get_tablet_replicas(manager, keyspace_name, table_name, token):
+async def get_tablet_replicas(manager: ManagerClient, server: ServerInfo, keyspace_name: str, table_name: str, token: int):
+    """
+    Gets tablet replicas of the tablet which owns a given token of a given table.
+    This call is guaranteed to see all prior changes applied to group0 tables.
+
+    :param server: server to query. Can be any live node.
+    """
+
+    host = manager.cql.cluster.metadata.get_host(server.ip_addr)
```

Comment:
I think Session.connect() waits for schema refresh.

Reply:
Cluster.connect() -> self.control_connection.connect() -> self._reconnect_internal() -> self._try_connect(host) -> self._refresh_node_list_and_token_map(connection, preloaded_results=shared_results)

Reply:
You put a lot of faith in the driver.
What if Cluster.connect() happened before the node booted?
But OK.

Reply:
> You put a lot of faith in the driver.
> What if Cluster.connect() happened before the node booted?
> But OK.

ManagerClient.server_add() calls _driver_update() after booting the node.

I think the wait for metadata to synchronize should belongs to connecting the driver and topology changes rather than every place which tries to access it.